libs/pulsar.patch

```diff
@@ -0,0 +1,1508 @@
++        return PULSAR_LZ4_read16(memPtr);
+     }
+     else
+     {
+@@ -162,9 +162,9 @@ static U16 LZ4_readLE16(const void* memPtr)
+     }
+ }
+ 
+-static void LZ4_writeLE16(void* memPtr, U16 value)
++static void PULSAR_LZ4_writeLE16(void* memPtr, U16 value)
+ {
+-    if (LZ4_isLittleEndian())
++    if (PULSAR_LZ4_isLittleEndian())
+     {
+         memcpy(memPtr, &value, 2);
+     }
+@@ -176,40 +176,40 @@ static void LZ4_writeLE16(void* memPtr, U16 value)
+     }
+ }
+ 
+-static U32 LZ4_read32(const void* memPtr)
++static U32 PULSAR_LZ4_read32(const void* memPtr)
+ {
+     U32 val32;
+     memcpy(&val32, memPtr, 4);
+     return val32;
+ }
+ 
+-static U64 LZ4_read64(const void* memPtr)
++static U64 PULSAR_LZ4_read64(const void* memPtr)
+ {
+     U64 val64;
+     memcpy(&val64, memPtr, 8);
+     return val64;
+ }
+ 
+-static size_t LZ4_read_ARCH(const void* p)
++static size_t PULSAR_LZ4_read_ARCH(const void* p)
+ {
+     if (LZ4_64bits())
+-        return (size_t)LZ4_read64(p);
++        return (size_t)PULSAR_LZ4_read64(p);
+     else
+-        return (size_t)LZ4_read32(p);
++        return (size_t)PULSAR_LZ4_read32(p);
+ }
+ 
+ 
+-static void LZ4_copy4(void* dstPtr, const void* srcPtr) { memcpy(dstPtr, srcPtr, 4); }
++static void PULSAR_LZ4_copy4(void* dstPtr, const void* srcPtr) { memcpy(dstPtr, srcPtr, 4); }
+ 
+-static void LZ4_copy8(void* dstPtr, const void* srcPtr) { memcpy(dstPtr, srcPtr, 8); }
++static void PULSAR_LZ4_copy8(void* dstPtr, const void* srcPtr) { memcpy(dstPtr, srcPtr, 8); }
+ 
+ /* customized version of memcpy, which may overwrite up to 7 bytes beyond dstEnd */
+-static void LZ4_wildCopy(void* dstPtr, const void* srcPtr, void* dstEnd)
++static void PULSAR_LZ4_wildCopy(void* dstPtr, const void* srcPtr, void* dstEnd)
+ {
+     BYTE* d = (BYTE*)dstPtr;
+     const BYTE* s = (const BYTE*)srcPtr;
+     BYTE* e = (BYTE*)dstEnd;
+-    do { LZ4_copy8(d,s); d+=8; s+=8; } while (d<e);
++    do { PULSAR_LZ4_copy8(d,s); d+=8; s+=8; } while (d<e);
+ }
+ 
+ 
+@@ -221,7 +221,7 @@ static void LZ4_wildCopy(void* dstPtr, const void* srcPtr, void* dstEnd)
+ #define COPYLENGTH 8
+ #define LASTLITERALS 5
+ #define MFLIMIT (COPYLENGTH+MINMATCH)
+-static const int LZ4_minLength = (MFLIMIT+1);
++static const int PULSAR_LZ4_minLength = (MFLIMIT+1);
+ 
+ #define KB *(1 <<10)
+ #define MB *(1 <<20)
+@@ -239,7 +239,7 @@ static const int LZ4_minLength = (MFLIMIT+1);
+ /**************************************
+ *  Common Utils
+ **************************************/
+-#define LZ4_STATIC_ASSERT(c)    { enum { LZ4_static_assert = 1/(int)(!!(c)) }; }   /* use only *after* variable declarations */
++#define LZ4_STATIC_ASSERT(c)    { enum { PULSAR_LZ4_static_assert = 1/(int)(!!(c)) }; }   /* use only *after* variable declarations */
+ 
+ 
+ /**************************************
+@@ -247,7 +247,7 @@ static const int LZ4_minLength = (MFLIMIT+1);
+ **************************************/
+ static unsigned LZ4_NbCommonBytes (register size_t val)
```

Comment:
I think this is also a function in librdkafka, so this also needs some `PULSAR_` prefix. I probably messed up something with the search before, sorry for that.

Reply:
nice catch!
This worked because it's static I think.